monai/data/dataset.py

```diff
@@ -120,8 +120,8 @@ def _pre_first_random_transform(self, item_transformed):
             random transform object
         """
         for _transform in self.transform.transforms:  # pytype: disable=attribute-error
-            # execute all the deterministic transforms before the first random transform
-            if isinstance(_transform, Randomizable):
+            # execute all the deterministic transforms
+            if isinstance(_transform, Randomizable) or not isinstance(_transform, MapTransform):
```

Comment:
why not just use `isinstance(_transform, Transform)`?

Reply:
I think we suppose the usage of CacheDataset is for `dict` transforms only so far.
Anyway, changing to `Transform` may be helpful for future extension, updated.
Thanks.